dali/operators/reader/video_reader_op_test.cc

```diff
@@ -248,4 +255,193 @@ TEST_F(VideoReaderTest, Vp9Profile2) {
   ASSERT_EQ(frames_shape[0][0], sequence_length);
 }
 
+TEST_F(VideoReaderTest, FrameLabels) {
+  const int sequence_length = 1;
+  const int iterations = 256;
+  const int batch_size = 1;
+  Pipeline pipe(batch_size, 1, 0);
+
+  pipe.AddOperator(
+    OpSpec("VideoReader")
+    .AddArg("device", "gpu")
+    .AddArg("random_shuffle", false)
+    .AddArg("sequence_length", sequence_length)
+    .AddArg("enable_frame_num", true)
+    .AddArg("image_type", DALI_YCbCr)
+    .AddArg(
+      "file_list", testing::dali_extra_path() +
+      "/db/video/frame_num_timestamp/file_list.txt")
+    .AddOutput("frames", "gpu")
+    .AddOutput("labels", "gpu")
+    .AddOutput("frame_num", "gpu"));
+
+  pipe.Build(this->Output_frames_label_frame_num());
+
+  DeviceWorkspace ws;
+  for (int i = 0; i < iterations; ++i) {
+    pipe.RunCPU();
+    pipe.RunGPU();
+    pipe.Outputs(&ws);
+    const auto &frames_gpu = ws.Output<dali::GPUBackend>(0);
+    const auto &label_gpu = ws.Output<dali::GPUBackend>(1);
+    const auto &frame_num_gpu = ws.Output<dali::GPUBackend>(2);
+
+    TensorList<CPUBackend> frames_cpu;
+    frames_cpu.Copy(frames_gpu, 0);
+    TensorList<CPUBackend> labels_cpu;
+    labels_cpu.Copy(label_gpu, 0);
+    TensorList<CPUBackend> frames_label_cpu;
+    frames_label_cpu.Copy(frame_num_gpu, 0);
+    cudaStreamSynchronize(0);
+
+    const uint8 *frames = static_cast<const uint8 *>(frames_cpu.raw_data());
+    const int *label = static_cast<const int *>(labels_cpu.raw_data());
+    const int *frames_label = static_cast<const int *>(frames_label_cpu.raw_data());
+
+    ASSERT_EQ(frames[0], frames_label[0]);
+  }
+}
+
+TEST_F(VideoReaderTest, FrameLabelsWithFileListFrameNum) {
```

Comment:
I would change this test to differ from the previous one. Because now enabling/disabling `file_list_frame_num` yields the same result so it is hard to tell if it works.